ymer/src/main/java/com/avanza/ymer/DocumentCollection.java

```diff
@@ -50,6 +53,17 @@
 
 	Stream<Document> findByQuery(Query query);
 
+	/**
+	 * Find by query and a given batch size. Data will be read from the underlying
+	 * mongo collection in batches of {@code batchSize} and returned as a stream of lists,
+	 * where each list is of (at most) {@code batchSize}
+	 *
+	 * @param query the query to filter by
+	 * @param batchSize used for MongoDB cursor and the size of the lists to return
+	 * @param includeFields fields to include. If not specified, all fields will be included. The {@code _id} field is always included.
+	 */
+	Stream<List<Document>> findByQuery(Query query, int batchSize, String... includeFields);
```

Comment:
The `Query` object includes a `fieldSpec` field whose intention seems to be to specify which fields to include or exclude. Another option than this `includeFields` parameter, would be to reuse that functionality in the `Query` object. Likewise, the `Query` object also contains a `cursorBatchSize` method.

Reply:
I have moved the functionality to the previously existing `findByQuery` method in https://github.com/AvanzaBank/ymer/pull/62/commits/803b343158653fe966908f698082d473c44a9606